File: algorithms/sieve-eratosthenes/sieve-eratosthenes.cpp
std::vector<int> sieve(int n) {
    std::vector<bool> numbers(n, true);
    std::vector<int> primes;
    for (int i = 1; i < n; i++) {
        if (i <= 1) {
            numbers[i] = false;
        }
        else {
            if (numbers[i]) {
                primes.push_back(i);
                for (int j = i; j < n; j += i) {
                    numbers[j] = false;
                }
            }
        }
    }

    return primes;
}

File: algorithms/sieve-eratosthenes/sieve-eratosthenes.py
def sieve(n):
    numbers = [True for _ in range(n)]
    primes = []
    for i in range(len(numbers)):
        if i <= 1:
            numbers[i] = False
        else:
            if numbers[i]:
                primes.append(i)
                for j in range(i, len(numbers), i):
                    numbers[j] = False

    return primes